src/main/kotlin/racingcar/Application.kt

```diff
@@ -1,5 +1,97 @@
 package racingcar
 
+import camp.nextstep.edu.missionutils.Console
+import camp.nextstep.edu.missionutils.Randoms
+
 fun main() {
-    // TODO: 프로그램 구현
+    println("경주할 자동차 이름을 입력하세요.(이름은 쉼표(,) 기준으로 구분)")
+    val inputCar = Console.readLine()
+    var cars = inputCar.split(",")
+    isValidCarName(cars)
+    cars = duplicateName(cars)
+
+    println("시도할 횟수는 몇 회인가요?")
+    val inputMove = Console.readLine()
+    val move = isMovementValid(inputMove)
+    race(cars, move)
+}
+
+fun isValidCarName(cars: List<String>){
+    isOver5Letters(cars)
+    isNameless(cars)
+}
+
+fun isOver5Letters(cars: List<String>) {
+    cars.forEach{ car ->
+        if (5 < car.length) {
+            throw IllegalArgumentException("이름이 5자를 넘습니다")
+        }
+    }
+}
+
+fun isNameless(cars: List<String>) {
+    cars.forEach{ car ->
+        if (car.isBlank()) {
+            throw IllegalArgumentException("이름이 없는 참가자가 있습니다")
+        }
+    }
+}
+
+fun duplicateName(cars: List<String>): List<String> {
+    val carAndNumber = mutableMapOf<String, Int>()
+    val result = mutableListOf<String>()
+
+    cars.forEach{car ->
+        if (carAndNumber.containsKey(car)) {
+            val carNumber = carAndNumber[car]!! + 1
+            carAndNumber[car] = carNumber
+            result.add("$car ${carNumber}호기")
+        } else {
+            carAndNumber[car] = 1
+            result.add(car)
+        }
+    }
+    return result
+}
```

Comment:
이런 처리 방법도 좋은 것 같아요!
다만 조금 코드가 길어지는 감이 있어서 `map`을 사용하시면 조금 더
간단하게 작성 가능할 것 같아요.

```suggestion
fun duplicateName(cars: List<String>): List<String> {
    val carMap = mutableMapOf<String, Int>()

    return cars.map { car ->
        val count = carMap.getOrDefault(car, 0) + 1
        carMap[car] = count
        if (count > 1) "$car ${count}호기" else car
    }
}
```

Reply:
- getOrDefault로 근본적인 값은 1로 설정하는 방법이 두 줄을 사용해서 1로 설정했던 제 방법과 비교해 좀더 간소하네요!
- 말씀하셨던 map이 key,value같은 map이 아니라 함수이름도 있다는것도 새롭게 알아갑니다.

### 질문
저는 return에 return 할 (여기서는 result)것만 쓰고, 그 result 를 구하는 로직은 return 위에 따로 작성하는게 보기 편하다고 생각해왔어요.
허나 구글링을 해보거나 ChatGPT를 써보거나 할때 return 에 result를 구하는 로직까지 첨가하는 사람들도 있더라구요.
**moondev03** 님께서는 제가 작성했던 코드를 보셨을때 보기 어려웠을까요?